Dev/GitHub/prnotii.5m.js

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env /usr/local/bin/node
+
+// <xbar.title>GitHub PRNotii</xbar.title>
+// <xbar.version>v1.0</xbar.version>
+// <xbar.author>aiden.ahn</xbar.author>
+// <xbar.author.github>eu81273</xbar.author.github>
+// <xbar.desc>Check Github Pull Reuqest</xbar.desc>
+// <xbar.image>https://github-production-user-asset-6210df.s3.amazonaws.com/2687916/246659270-e08d6380-62f9-478b-b35b-e2222cf65791.png</xbar.image>
+// <xbar.var>string(VAR_TOKEN=""): GitHub personal access token with repos scope</xbar.var>
+// <xbar.var>string(VAR_ENDPOINT="https://api.github.com"): GitHub API endpoint</xbar.var>
+
+// <swiftbar.hideAbout>true</swiftbar.hideAbout>
+// <swiftbar.hideRunInTerminal>true</swiftbar.hideRunInTerminal>
+// <swiftbar.hideLastUpdated>false</swiftbar.hideLastUpdated>
+// <swiftbar.hideDisablePlugin>true</swiftbar.hideDisablePlugin>
+// <swiftbar.hideSwiftBar>true</swiftbar.hideSwiftBar>
+
+const https = require('https');
+const {join} = require('path');
+const {homedir} = require('os');
+const {readFile} = require('fs/promises');
+
+const request = (url, headers) => new Promise((resolve, reject) => {
+  const TIMEOUT_DELAY = 5000; // 5 Sec
+  const body = [];
+  const req = https.get(url, {headers}, res => {
+    res.on('data', chunk => body.push(chunk));
+    res.on('end', () => resolve(Buffer.concat(body).toString()));
+  });
+  req.on('error', reject);
+  req.end();
+  setTimeout(reject, TIMEOUT_DELAY, new Error('PRNOTII_ETIMEDOUT'));
+});
+const refine = items => items.map(item => ({
+  repository: item.repository_url.replace(/.+\//, ''),
+  href: item.html_url,
+  title: item.title,
+}));
+const organize = refined => refined.reduce((prev, item) => {
+  prev[item.repository] = prev[item.repository] || [];
+  prev[item.repository].push(item);
+  return prev;
+}, {});
+const githubQuery = (endpoint, token, query) => 
+  request(endpoint + '/search/issues?q=' + encodeURIComponent(query), {'Authorization': 'token ' + token})
```

Comment:
@eu81273 

Reading in the configuration is actually working fine. The error I'm encountering (`Unexpected token R in JSON at position 2`) is this line. The GitHub API requires a user agent sent in the headers. This cleared things up for me:

```suggestion
  request(endpoint + '/search/issues?q=' + encodeURIComponent(query), {'User-Agent': 'xbar-app', 'Authorization': 'token ' + token})
```

I'll leave it to you to decide what you might want to put in as the user agent. You could promote your work in -- `prnotti-for-xbar-by-eu81273`. 😄